Call `GapObj` instead of accessing `chi.values` (for group characters) or `f.map` (for group homomorphisms)

ThomasBreuer opened the pull request:
This is a follow-up of #3671.

ThomasBreuer committed:
use `GapObj` instead of `.values` access for group characters

ThomasBreuer committed:
use `GapObj(f)` not `f.map` for `f::GAPGroupHomomorphism`

(and replace more `.X` access by `GapObj`)

ThomasBreuer committed:
replace `GAPTable` for char. tables by `GapObj`

fingolfin commented on src/Groups/group_characters.jl:
```diff
@@ -3045,8 +3047,8 @@ function symmetric_parts(characters::Vector{GAPGroupClassFunction}, n::Int)
-            for chi in GAPWrap.SymmetricParts(GAPTable(tbl),
-                         GAP.GapObj([chi.values for chi in characters]), n)]
+            for chi in GAPWrap.SymmetricParts(GapObj(tbl),
+                         GapObj([GapObj(chi) for chi in characters]), n)]
```
For this kind of conversion you could now also write this:
```suggestion
                         GapObj(characters; recursive=true), n)]
```

fingolfin commented on src/Groups/homomorphisms.jl:
```diff
@@ -118,12 +118,12 @@ If `check` is set to `false` then it is not checked whether the mapping
-  vgens = GapObj([x.X for x in gensG])
-  vimgs = GapObj([x.X for x in imgs])
+  vgens = GapObj([GapObj(x) for x in gensG])
+  vimgs = GapObj([GapObj(x) for x in imgs])
```
Here's another place one could now just do a recursive conversion:
```suggestion
  vgens = GapObj(gensG; recursive=true)
  vimgs = GapObj(imgs; recursive=true)
```

ThomasBreuer committed:
simplify recursive `GapObj` calls

fingolfin commented on src/Groups/group_characters.jl:
```diff
@@ -2182,14 +2188,14 @@ function induce(chi::GAPGroupClassFunction, tbl::GAPGroupCharacterTable)
-  ind = GAPWrap.InducedClassFunctionsByFusionMap(GAPTable(parent(chi)),
-          GAPTable(tbl), GapObj([chi.values]), GapObj(fusion))
+  ind = GAPWrap.InducedClassFunctionsByFusionMap(GapObj(parent(chi)),
+          GapObj(tbl), GapObj([GapObj(chi)]), GapObj(fusion))
```
```suggestion
          GapObj(tbl), GapObj([chi]; recursive=true), GapObj(fusion))
```

fingolfin commented on src/Groups/group_characters.jl:
```diff
@@ -2432,13 +2438,13 @@ coordinates(chi::GAPGroupClassFunction) = coordinates(QQFieldElem, chi)
     if characteristic(t) == 0
       # use scalar products for an ordinary character
-      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([chi.values]))
+      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([GapObj(chi)]))
```
```suggestion
      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([chi]; recursive=true))
```

fingolfin commented on src/Groups/group_characters.jl:
```diff
@@ -2432,13 +2438,13 @@ coordinates(chi::GAPGroupClassFunction) = coordinates(QQFieldElem, chi)
     else
       # decompose a Brauer character
-      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([chi.values]), GapObj("nonnegative"))
+      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([GapObj(chi)]), GapObj("nonnegative"))
```
```suggestion
      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([chi]; recursive=true), GapObj("nonnegative"))
```

ThomasBreuer committed:
more `GapObj` calls using `recursive = true`

ThomasBreuer committed:
preliminary hack for GAP.jl

add a `GAP.julia_to_gap(::JSON3.Array, ...)` method
until the fix from oscar-system/GAP.jl/pull/989 is available

ThomasBreuer committed:
add a missing `using` statement

ThomasBreuer commented:
failure in `test (1.10, short, macOS, RPTU)`:
```
ERROR: LoadError: On worker 2:
Failed to precompile Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13] to "/Users/aaruni/.julia/compiled/v1.10/Oscar/jl_VGgVXt".
```

lgoettgens commented:
> failure in `test (1.10, short, macOS, RPTU)`:
> 
> ```
> ERROR: LoadError: On worker 2:
> Failed to precompile Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13] to "/Users/aaruni/.julia/compiled/v1.10/Oscar/jl_VGgVXt".
> ```

This type of failure is currently discussed on slack. Restarting the job seemed to work in this particular case
